host/xtest/regression_1000.c

```diff
@@ -2036,3 +2037,190 @@ static void xtest_tee_test_1025(ADBG_Case_t *c)
 }
 ADBG_CASE_DEFINE(regression, 1025, xtest_tee_test_1025,
 		 "Test memref NULL and/or 0 bytes size");
+
+#define TEE_UUID_NS_NAME_SIZE  128
+
+/*
+ * TEE Client UUID name space identifier (UUIDv4)
+ *
+ * Value here is random UUID that is allocated as name space identifier for
+ * forming Client UUID's for TEE environment using UUIDv5 scheme.
+ */
+static const char *client_uuid_linux_ns = "58ac9ca0-2086-4683-a1b8-ec4bc08e01b6";
+
+/* TEEC_LOGIN_PUBLIC's Client UUID is NIL UUID */
+static TEEC_UUID client_uuid_public = { };
+
+static TEEC_Result get_client_identity(TEEC_Session *session, uint32_t *login,
+				       TEEC_UUID *client_uuid)
+{
+	TEEC_Operation operation = { };
+	TEEC_Result result = TEEC_ERROR_GENERIC;
+
+	operation.params[1].tmpref.buffer = client_uuid;
+	operation.params[1].tmpref.size = sizeof(*client_uuid);
+
+	operation.paramTypes = TEEC_PARAM_TYPES(TEEC_VALUE_OUTPUT,
+						TEEC_MEMREF_TEMP_OUTPUT,
+						TEEC_NONE, TEEC_NONE);
+
+	result = TEEC_InvokeCommand(session, TA_OS_TEST_CMD_CLIENT_IDENTITY,
+				    &operation, NULL);
+
+	if (result != TEEC_SUCCESS)
+		return result;
+
+	*login = operation.params[0].value.a;
+
+	return TEEC_SUCCESS;
+}
+
+static void xtest_tee_test_1026(ADBG_Case_t *c)
+{
+	TEEC_Result result = TEEC_ERROR_GENERIC;
+	uint32_t ret_orig = 0;
+	TEEC_Context context = { };
+	TEEC_Session session = { };
+	uint32_t login = (uint32_t)-1;
+	TEEC_UUID client_uuid = { };
+
+	result = TEEC_InitializeContext(NULL, &context);
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result))
+		return;
+
+	result = TEEC_OpenSession(&context, &session, &os_test_ta_uuid,
+				  TEEC_LOGIN_PUBLIC, NULL, NULL, &ret_orig);
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result))
+		goto cleanup_context;
+
+	result = get_client_identity(&session, &login, &client_uuid);
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result))
+		goto cleanup_session;
+
+	ADBG_EXPECT_COMPARE_UNSIGNED(c, login, ==, TEEC_LOGIN_PUBLIC);
+
+	ADBG_EXPECT_EQUAL(c, &client_uuid_public, &client_uuid,
+			  sizeof(TEEC_UUID));
+
+cleanup_session:
+	TEEC_CloseSession(&session);
+
+cleanup_context:
+	TEEC_FinalizeContext(&context);
+}
+
+ADBG_CASE_DEFINE(regression, 1026, xtest_tee_test_1026,
+		 "Session: public login");
+
+static void xtest_tee_test_1027(ADBG_Case_t *c)
+{
+	TEEC_Result result = TEEC_ERROR_GENERIC;
+	uint32_t ret_orig = 0;
+	TEEC_Context context = { };
+	TEEC_Session session = { };
+	uint32_t login = (uint32_t)-1;
+	TEEC_UUID client_uuid = { };
+	TEEC_UUID expected_client_uuid = { };
+	TEEC_UUID uuid_ns = { };
+	char uuid_name[TEE_UUID_NS_NAME_SIZE];
+
+	result = xtest_uuid_from_str(&uuid_ns, client_uuid_linux_ns);
+
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result))
+		return;
+
+	sprintf(uuid_name, "uid=%x", geteuid());
+
+	result = xtest_uuid_v5(&expected_client_uuid, &uuid_ns, uuid_name,
+			       strlen(uuid_name));
+	if (!ADBG_EXPECT_TEEC_SUCCESS(c, result))
+		return;
+
+	result = TEEC_InitializeContext(NULL, &context);
```

Comment:
It's OK to rely on `xtest_teec_ctx` being properly initialized here.

Reply:
ok

Reply:
updated